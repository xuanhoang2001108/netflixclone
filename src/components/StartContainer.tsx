import AppBar from "@mui/material/AppBar";
import { APP_BAR_HEIGHT } from "../constant";
import Stack from "@mui/material/Stack";
import TextField from "@mui/material/TextField";
import { StartHeader } from "./StartHeader";
import { useNavigate } from "react-router-dom";

const StartContainer = () => {
  const navigate = useNavigate();
  const handleNavigateToLogin = () => {
    navigate("/LoginPage");
  };
  return (
    <div className="justify-center">
      <AppBar
        sx={{
          height: APP_BAR_HEIGHT,
          backgroundImage: "none",
          boxShadow: 0,
        }}
      >
        <img
          className="w-screen h-screen"
          src="src/assets/background-image.jpg"
          alt="Background"
          style={{
            objectFit: "cover",
            position: "absolute",
          }}
        />
        <StartHeader />
        <div className="absolute  w-screen h-screen text-center mt-72">
          <h1 className="font-bold text-5xl">
            Unlimited movies, TV shows, and more
          </h1>
          <h1 className=" text-2xl mt-6">Watch anywhere. Cancel anytime</h1>
          <h1 className=" text-2xl mt-6">
            Ready to watch? Enter your email to create or restart your
            membership.
          </h1>
          <div className="flex justify-center">
            <Stack direction={"row"} sx={{ marginTop: "18px" }} spacing={1}>
              <TextField
                label="Email Address"
                variant="outlined"
                style={{
                  width: 400,
                  backgroundColor: "rgba(61, 66, 61, 0.3)",
                }}
              />

              <button
                className="rounded-md bg-red-600 w-44 h-18 text-2xl font-bold"
                onClick={handleNavigateToLogin}
              >
                Get Started
              </button>
            </Stack>
          </div>
        </div>
      </AppBar>
    </div>
  );
};

export default StartContainer;
