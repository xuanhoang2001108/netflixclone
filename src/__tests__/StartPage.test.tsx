import { render, screen, fireEvent } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import renderer from 'react-test-renderer';
import StartContainer from '../components/StartContainer';

const mockNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockNavigate,
}));

describe('StartContainer', () => {
  it('should navigate to /LoginPage when Get Started button is clicked', () => {
    // Render the component with MemoryRouter
    render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path="/" element={<StartContainer />} />
        </Routes>
      </MemoryRouter>
    );

    // Click the Get Started button
    fireEvent.click(screen.getByText('Get Started'));

    // Check if the navigate function was called with the correct argument
    expect(mockNavigate).toHaveBeenCalledWith('/LoginPage');

    // Snapshot testing
    const tree = renderer.create(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path="/" element={<StartContainer />} />
        </Routes>
      </MemoryRouter>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
