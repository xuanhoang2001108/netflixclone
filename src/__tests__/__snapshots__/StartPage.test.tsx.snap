// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartContainer should navigate to /LoginPage when Get Started button is clicked 1`] = `
<div
  className="justify-center"
>
  <header
    className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-12g3m0k-MuiPaper-root-MuiAppBar-root"
  >
    <img
      alt="Background"
      className="w-screen h-screen"
      src="/assets/background-image.jpg"
      style={
        {
          "objectFit": "cover",
          "position": "absolute",
        }
      }
    />
    <div
      className="absolute w-screen  flex flex-row items-center  place-content-between"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Netflix Logo"
          className="MuiBox-root css-1ujvwfr"
          src="/assets/netflix-logo.png"
        />
      </a>
      <button
        className="rounded-md bg-red-600 w-20 h-8 font-semibold mr-40 items-center"
        onClick={[Function]}
      >
        Sign In
      </button>
    </div>
    <div
      className="absolute  w-screen h-screen text-center mt-72"
    >
      <h1
        className="font-bold text-5xl"
      >
        Unlimited movies, TV shows, and more
      </h1>
      <h1
        className=" text-2xl mt-6"
      >
        Watch anywhere. Cancel anytime
      </h1>
      <h1
        className=" text-2xl mt-6"
      >
        Ready to watch? Enter your email to create or restart your membership.
      </h1>
      <div
        className="flex justify-center"
      >
        <div
          className="MuiStack-root css-1krzmza-MuiStack-root"
        >
          <div
            className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            style={
              {
                "backgroundColor": "rgba(61, 66, 61, 0.3)",
                "width": 400,
              }
            }
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-sizeMedium MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor=":r0:"
              id=":r0:-label"
            >
              Email Address
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id=":r0:"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Email Address
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <button
            className="rounded-md bg-red-600 w-44 h-18 text-2xl font-bold"
            onClick={[Function]}
          >
            Get Started
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
